runtime/native.go

```diff
@@ -348,6 +360,21 @@ func getNativeType(rtype reflect.Type) *Type {
 	return t
 }
 
+func newNativeField(name string, i int) *Object {
+	nativeFieldGet := func(f *Frame, args Args, _ KWArgs) (*Object, *BaseException) {
+		if raised := checkFunctionArgs(f, name, args, nativeType); raised != nil {
```

Comment:
This check needs to verify that arg[0] is the particular type being created, not just any native type. If this getter is created for *SomeStruct and I call fget with, say, *int then it will pass this check (it is a native type) but the v.Field() call below will panic.

This may require a small reorganization where you create the type before fully populating its dict in getNativeType() above. Something like:

```
t = &newNativeType(rtype, base, newStringDict(d)).Type
for each field {
  raised := t.dict.SetItemString(f, fieldName, newNativeField(fieldName, fieldIndex, t))
  ...
}
```